.github/workflows/windows-meson-msys2.yml

```diff
@@ -0,0 +1,136 @@
+name: MSYS2 builds
+
+on: [push, pull_request]
+
+jobs:
+  build_win_meson_msys2_release:
+    name: ${{ matrix.conf.name }}
+    runs-on: windows-latest
+    if: github.event_name != 'pull_request' || contains('dreamer,kcgen,ant-222,Wengier,shermp', github.actor) == false
+    
+    strategy:
+      matrix:
+        conf:
+          - name: Meson MinGW 32-bit
+            arch: i686
+            sys: MINGW32
+          - name: Meson MinGW 64-bit
+            arch: x86_64
+            sys: MINGW64
+    
+    defaults:
+      run:
+        shell: msys2 {0}
+    
+    steps:
+      - name: Checkout repository and submodules
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{matrix.conf.sys}}
+          update: true
+          install: >-
+            git
+            mingw-w64-${{matrix.conf.arch}}-meson
+            mingw-w64-${{matrix.conf.arch}}-gcc
+            mingw-w64-${{matrix.conf.arch}}-ccache
+            mingw-w64-${{matrix.conf.arch}}-pkgconf
+            mingw-w64-${{matrix.conf.arch}}-ntldd
+            mingw-w64-${{matrix.conf.arch}}-ncurses
+            mingw-w64-${{matrix.conf.arch}}-glib2
+            mingw-w64-${{matrix.conf.arch}}-fluidsynth
+            mingw-w64-${{matrix.conf.arch}}-munt-mt32emu
+            mingw-w64-${{matrix.conf.arch}}-libpng
+            mingw-w64-${{matrix.conf.arch}}-opusfile
+            mingw-w64-${{matrix.conf.arch}}-SDL2
+            mingw-w64-${{matrix.conf.arch}}-SDL2_net
+            mingw-w64-${{matrix.conf.arch}}-zlib
+      
+      - name: Log environment
+        run:  ./scripts/log-env.sh
+
+      - name: Inject version string
+        run: |
+          set -x
+          git fetch --prune --unshallow
+          export VERSION=$(git describe --abbrev=5)
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+      
+      - name: Setup release build
+        run: |
+          meson setup \
+            -Dbuildtype=release \
+            -Ddefault_library=static \
+            -Db_asneeded=true \
+            -Dtry_static_libs=png,opusfile \
+            build
+      
+      - name: Build
+        run: ninja -C build
+
+      - name: Strip binary
+        run: strip build/dosbox.exe
+      
+      - name: Package
+        run: |
+          set -x
+          mkdir -p dest/doc
+
+          cp build/dosbox.exe      dest/
+          cp COPYING               dest/COPYING.txt
+          cp docs/README.template  dest/README.txt
+          cp docs/README.video     dest/doc/video.txt
+          cp README                dest/doc/manual.txt
+
+
+          # The list of required dll files and their source paths can be found using ntldd
+          for lib in $(ntldd -R dest/dosbox.exe | sed -e 's/^[[:blank:]]*//g' | cut -d ' ' -f 3 | grep -E -i 'mingw(32|64)' | sed -e 's|\\|/|g'); do 
+            cp "$lib" dest/
+          done
```

Comment:
Would be nice to wrap the dll discovery and cleanup code. 
Maybe xargs at the end could simplify it, too?

``` shell
          ntldd -R dest/dosbox.exe \
            | sed -e 's/^[[:blank:]]*//g' \
            | cut -d ' ' -f 3 \
            | grep -E -i 'mingw(32|64)' \
            | sed -e 's|\\|/|g' \
            | xargs cp --target-directory=dest/
```

Reply:
That should work (tested locally). My shell knowledge isn't always up to snuff. I'm a Windows guy who uses linux, rather than the other way around.